### DISCUSSION TITLE
Setting maxWaitMillis from 50ms to 25ms causes a large number of cluster slot commands to be executed

### DISCUSSION THREAD

**Discussion by bozai111** [xetorthio/jedis#3379] [Q&A]
Redis cluster and redis client version information：
jedis version :2.9
Redis cluster version : 4.0   Shard deployment：100 master  100 slave  qps：2w per second for a single shard

scene description
Setting maxWaitMillis and soTimeout parameter from 50ms to 25ms causes a large number of cluster slot commands to be executed,
The redis cluster load is very high and the cluster service is unavailable,Is there a best practice for jedis parameter setting?

**Comment by sazzad16** [comment]
First of all, Jedis 2.9 an ancient version. Please use a latest/updated release.
There is no defined 'best' practice but it'll be a good idea to increase the time related parameters. I'm assuming your maxTotal is not high enough, so increase that as well.

**Reply by bozai111** [reply]
The redis cluster currently has 100 shards, 100 master nodes and 100 slave nodes. A single redis cluster is very large, with a data volume of about 1TB. The number of connections in a single shard is about 10,000-20,000, and there are about 2,000 application servers. , so maxTotal is set to a relatively large value, and the number of connections on the redis cluster side will be large, which will occupy redis memory. Is there a recommended value for maxTotal?

**Reply by sazzad16** [reply]
> Is there a recommended value for maxTotal?

No. commons pool uses `8` as default and it seems _good enough_ for most applications. But as it seems it's not good enough for you.

**Reply by bozai111** [reply]
Current development is used to expanding the number of application servers to accommodate more business requests, and the number of connections to redis cluster is also increasing. Do we need to restrict this behavior?

**Reply by sazzad16** [reply]
Is Jedis(Cluster) running in application servers?

**Reply by bozai111** [reply]
yes，There is jedis on each application server, and the jedis client will be initialized when the application service is started

**Reply by sazzad16** [reply]
> Current development is used to expanding the number of application servers to accommodate more business requests, and the number of connections to redis cluster is also increasing. Do we need to restrict this behavior?

There's no simple answer. If you find Redis servers (shards) are getting very busy, you may consider the restriction.
